haskell/src/Monocle/Version.hs

```diff
@@ -0,0 +1,16 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+-- | Monocle version
+module Monocle.Version (version) where
+
+import Data.Version (showVersion)
+import Development.GitRev (gitDirty, gitHash)
+import qualified Paths_monocle as P (version)
+import Relude
+
+version :: String
+version = concat [showVersion P.version, " ", $(gitHash), dirty]
+  where
+    dirty
+      | $(gitDirty) = "-dirty"
+      | otherwise = ""
```

Comment:
for th-env:
```suggestion
import Language.Haskell.TH.Env (envQ)

version :: String
version = concat [showVersion P.version, " ", commit]
  where
    commit = case fromEnv of
      Just v -> v
      Nothing -> $(gitHash) <> dirty
    fromEnv :: Maybe String
    fromEnv = $$(envQ "MONOCLE_COMMIT")
    dirty
      | $(gitDirty) = "-dirty"
      | otherwise = ""
```